atlasdb-dbkvs/src/main/java/com/palantir/atlasdb/keyvalue/dbkvs/impl/oracle/OracleDdlTable.java

```diff
@@ -118,6 +123,33 @@ public void create(byte[] tableMetadata) {
                 "INSERT INTO " + config.metadataTable().getQualifiedName() + " (table_name, table_size) VALUES (?, ?)");
     }
 
+    private void maybeAlterTableToHaveOverflow() {
+        try {
+            String shortTableName = oracleTableNameGetter.getInternalShortTableName(conns, tableRef);
+
+            if (config.alterTablesOrMetadataToMatch().contains(tableRef)
+                    && overflowTableHasMigrated()
+                    && overflowTableExists()
+                    && !overflowColumnExists(shortTableName)) {
+                alterTableToHaveOverflowColumn(shortTableName);
+            }
+        } catch (TableMappingNotFoundException e) {
+            log.warn("Table mapping expected but not found for table.", UnsafeArg.of("tableRef", tableRef), e);
+        }
+    }
+
+    private void alterTableToHaveOverflowColumn(String shortTableName) {
+        log.info(
+                "Altering table to have overflow column to match metadata.",
+                UnsafeArg.of("tableReference", tableRef),
+                UnsafeArg.of("shortTableName", shortTableName));
+        conns.get().executeUnregisteredQuery("ALTER TABLE " + shortTableName + " ADD (overflow NUMBER(38))");
+    }
+
+    private boolean overflowTableHasMigrated() {
+        return config.overflowMigrationState() != OverflowMigrationState.UNSTARTED;
```

Comment:
Is there some context on why this shouldn't be finished? (i.e., why is in progress a valid state to do this in?)

Reply:
It's probably safe, I just didn't want to have to prove that it is (another test case/etc)! Plus, no should be using the old table anyways, and all new tables moving forward should already be migrated.

Reply:
Sorry, my question is more - shouldn't  we err on the side of caution here and only do modifications in the FINISHED state, rather than allowing modifications in IN_PROGRESS, FINISHING and FINISHED?

Reply:
I was going off of lines 107-109, where the overflow table is only created as long as the state is not `UNSTARTED`, but agreed, let's err on the side of caution here.